session.go

```diff
@@ -80,11 +175,31 @@ func (p *ProxyListener) StartListeners() {
 
 	// Previously authenticated listeners can be any network type
 	// including UNIX domain sockets.
-	p.InitAuthenticatedListeners()
-	for _, listener := range p.cfg.Listeners {
-		p.wg.Add(1)
-		go p.FilterAcceptLoop(listener.Net, listener.Address)
+	p.initAuthenticatedListeners()
+
+	// start the main listeners
+	handleNewConnection := func(conn net.Conn) error {
```

Comment:
Perhaps to be consistent with how the auth'ed listeners are initialized, perhaps we should also break this out into a separate function?